pkg/util/tests/kubectl/kubectl.go

```diff
@@ -0,0 +1,274 @@
+	}
+
+	return resourceList, nil
+}
+
+// GetResource returns the json of a resource
+func GetResource(ns, resourceType, resourceName string) (string, error) {
+	cmd := kubectl("get %s %s -n %s -o json", resourceType, resourceName, ns)
+	json, err := shell.ExecuteCommand(cmd)
+	if err != nil {
+		return "", err
+	}
+
+	return json, nil
+}
+
+// GetPodFromLabel returns the pod name from a label, if there is more than one pod, it will return an error
+func GetPodFromLabel(ns, label string) (string, error) {
+	var podList []string
+	podList, err := GetPodsFromLabel(ns, label)
+	if err != nil {
+		return "", err
+	}
+	if len(podList) > 1 {
+		return "", fmt.Errorf("more than one pod found with label %s", label)
+	}
+	if len(podList) == 0 {
+		return "", fmt.Errorf("no pod found with label %s", label)
+	}
+
+	return podList[0], nil
+}
+
+// GetPodsFromLabel returns the pod name from a label
+func GetPodsFromLabel(ns, label string) ([]string, error) {
+	var podList []string
+	cmd := kubectl("get pods -n %s -l %s -o jsonpath={.items[*].metadata.name}", ns, label)
+	output, err := shell.ExecuteCommand(cmd)
+	podList = strings.Split(output, " ")
+	if err != nil {
+		return podList, fmt.Errorf("error getting pods names: %v, output: %s", err, output)
+	}
+	return podList, nil
+}
+
+// CreateNamespace creates a namespace
+// If the namespace already exists, it will return nil
+func CreateNamespace(ns string) error {
+	cmd := kubectl("create namespace %s", ns)
+	output, err := shell.ExecuteCommand(cmd)
+	if err != nil {
+		if strings.Contains(output, "AlreadyExists") {
+			return nil
+		}
+
+		return err
+	}
+
+	return nil
+}
+
+// DeleteNamespace deletes a namespace
+func DeleteNamespace(ns string) error {
+	cmd := kubectl("delete namespace %s", ns)
+	_, err := shell.ExecuteCommand(cmd)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// CheckNamespaceExist checks if a namespace exists
+// If the namespace exists, it will return nil
+// If the namespace does not exist, it will return an error
+func CheckNamespaceExist(ns string) error {
+	cmd := kubectl("get namespace %s -o jsonpath={metadata.name}", ns)
+	_, err := shell.ExecuteCommand(cmd)
+	if err != nil {
+		if strings.Contains(err.Error(), "not found") {
+			return ErrNotFound
+		}
+
+		return err
+	}
+
+	return nil
+}
+
+// GetDeployments returns the deployments of a namespace
+func GetDeployments(ns string) ([]string, error) {
+	var deployments []string
+	cmd := kubectl("get deployments -n %s -o jsonpath={.items[*].metadata.name}", ns)
+	output, err := shell.ExecuteCommand(cmd)
+	deployments = strings.Split(output, " ")
+	if err != nil {
+		return deployments, fmt.Errorf("error getting deployments names: %v, output: %s", err, output)
+	}
+	return deployments, nil
+}
+
+// GetPodLogs returns the logs of a deployment
+// Arguments:
+// - ns: namespace
+// - deploymentName: deployment name
+// - Since: time range
+func GetPodLogs(ns, podName, since string) (string, error) {
```

Comment:
```suggestion
func GetPodLogs(ns, podName, since time.Duration) (string, error) {
```

Reply:
Or even replace `since` with `options ...any` and then check if any of the provided options is a `time.Duration`. This will allow us to call `GetPodLogs` without specifying `since`. But maybe we'll realize that we should always specify it.